utils/s2n_safety.h

```diff
@@ -100,7 +100,7 @@ static inline void* trace_memcpy_check(void *restrict to, const void *restrict f
 #define GUARD_NONNULL_PTR( x )          do {if ( (x) == NULL ) return NULL;} while (0)
 
 /* Check the return value from caller. If this value is -2, S2N_ERR_BLOCKED is marked*/
-#define GUARD_AGAIN( x )  do {if ( (x) == -2 ) { S2N_ERROR(S2N_ERR_BLOCKED); } GUARD( x );} while(0)
+#define GUARD_AGAIN( x )  do {if ( (x) == S2N_ERR_CACHE_WILL_BLOCK ) { S2N_ERROR(S2N_ERR_BLOCKED); } GUARD( x );} while(0)
```

Comment:
Can we not make this a widely available GUARD? It seems unintuitive and is only used in one specific situation. I'd prefer we just handle that situation with an explicit if statement.

Reply:
Makes sense. Added a special case in the cache code.